src/client_builder.rs

```diff
@@ -61,35 +129,88 @@ where
         ClientBuilder {
             domain,
             port,
-            starttls: false,
+            mode: ConnectionMode::Tcp,
+            tls_kind: TlsKind::Any,
+            no_tls_verify: false,
         }
     }
 
     /// Use [`STARTTLS`](https://tools.ietf.org/html/rfc2595) for this connection.
-    #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
```

Comment:
We should still require one of these features to be set, because otherwise `ConnectionMode::StartTls` won't work.

Reply:
Which I handled in the connection method by having it return an error staging that no TLS was configured.

Any the reason I removed this #[cfg] was that if I didn't this enumeration would have zero values when no TLS implementation was compiled in.   Leaving this in felt the simplest way to handle that nicely.

the other way would be to completely CFG out this entire enum and tlskind struct member and other functions. (which I can do instead if you prefer that rough)

Reply:
Why is it a problem for the enum to have no variants?

I'd rather have things fail at compile-time (i.e., make it impossible to write code that tries to `STARTTLS` when there's no TLS available), rather than have it return a runtime error.

Reply:
I removed the entire type when no TLS backend is compiled in, and the tls/starttls methods are also removed (based on your later comments).  So the runtime error code is gone now.